src/unitxt/operators.py

```diff
@@ -671,6 +674,100 @@ def process(self, stream: Stream, stream_name: str = None) -> Generator:
+    When 'min_frequency_percent' is larger than 0, remove from the remaining list any value whose relative frequency makes
+    less than 'min_frequency_percent' of the total number of rows in the stream.
+
+    Examples:
+
+    ExtractFieldValues(stream_name="train", field="label", to_field="classes") - extracts all the unique values of
+    field "label" and stores in field "classes" of each and every record in all streams.
+
+    ExtractFieldValues(stream_name="train", field="labels", to_field="classes", process_every_value=True) -
+    in case that field 'labels' contains a list of values (and not a single value) - track the occurrences of all the possible
+    values in these lists, and report the most frequent values.
+    if process_every_value=False, track the most frequent whole lists, and report those (as a list of lists) in field 'to_field'
+
+    ExtractFieldValues(stream_name="train", field="label", to_field="classes",overall_top_frequency_percent=80) -
+    extracts the most frequent possible values of field 'label' that cover at least 80% of the samples,
+    and stores them in field 'classes' of all streams.
+
+    ExtractFieldValues(stream_name="train", field="label", to_field="classes",min_frequency_percent=5) -
+    extracts all possible values of field 'label' that cover, each, at least 5% of the instances.
+    Stores these values, sorted by decreasing order of frequency, in field 'classes' of each record in all streams.
+    """
+
+    def process(self, multi_stream: MultiStream) -> MultiStream:
+        stream = multi_stream[self.stream_name]
+        all_values = []
+        for instance in stream:
+            if (not isinstance(instance[self.field], list)) or (self.process_every_value == False):
+                all_values.append(
+                    (*instance[self.field],) if isinstance(instance[self.field], list) else instance[self.field]
+                )  # convert to a tuple if list, to enable Counter
+            else:
+                all_values.extend(instance[self.field])
+        counter = Counter(all_values)
+        values_and_counts = counter.most_common()
+        if self.overall_top_frequency_percent < 100:
+            top_frequency = math.ceil(len(all_values) * self.overall_top_frequency_percent / 100)
+            sum_counts = 0
+            for i, p in enumerate(values_and_counts):
+                sum_counts += p[1]
+                if sum_counts >= top_frequency:
+                    break
+            values_and_counts = counter.most_common(i + 1)
+        if self.min_frequency_percent > 0:
+            min_frequency = math.floor(self.min_frequency_percent * len(all_values) / 100)
+            while values_and_counts[-1][1] < min_frequency:
+                values_and_counts.pop()
+        values_to_keep = [[*ele[0]] if isinstance(ele[0], tuple) else ele[0] for ele in values_and_counts]
+        if self.to_field is None:
```

Comment:
I suggest we just throw an error if to_field is not set.  It's more consistent with other operators.

Reply:
I accept.